node/connection.js

```diff
@@ -105,6 +105,10 @@ TChannelConnection.prototype.setupSocket = function setupSocket() {
         }));
 
         if (self.ephemeral) {
+            var peer = self.channel.peers.get(self.socketRemoteAddr);
+            if (peer) {
+                peer.close(noop);
+            }
```

Comment:
This is great and fits our immediate need, but you might be able to knock a non-ephemeral peer out by issuing an ephemeral request from it.